Extensions/xkit_patches.js

```diff
@@ -220,12 +266,18 @@ XKit.extensions.xkit_patches = new Object({
 				function handler(e) {
 					if (e.origin === window.location.protocol + "//" + window.location.host && e.data.timestamp === "xkit_" + details.timestamp) {
 						window.removeEventListener("message", handler);
-						let {success, response} = e.data;
+						let {success, response} = JSON.parse(JSON.stringify(e.data));
 
 						if (typeof response.headers["x-tumblr-kittens"] !== "undefined") {
 							XKit.interface.kitty.set(response.headers["x-tumblr-kittens"]);
 						}
 
+						response.json = function() {
+							try {
+								return JSON.parse(response.responseText);
+							} catch (err) { return null; }
```

Comment:
part of the idea of moving it into a function is that we don't need to swallow errors now (so if we're expecting json and don't get it, things make more sense)

Reply:
i cant tell what youre thinking of here? i cant think of anything more useful to return and the error message from a failed `JSON.parse` isnt exactly ever useful

Reply:
I just don't think it makes any sense to explicitly swallow this error when we have no reason to or expectation that it will ever fail.

Reply:
we can expect it to fail though - when requests to json endpoints are rejected, sometimes they return useful information, but sometimes they return html error pages (most easily observed if you block referers). if we want to use that information we would have to explicitly use try/catch every time when there's no benefit to actually catching the parsing error.